llvm/lib/Target/RISCV/MCTargetDesc/RISCVAsmBackend.cpp

```diff
@@ -175,24 +181,38 @@ void RISCVAsmBackend::relaxInstruction(MCInst &Inst,
     Res.addOperand(Inst.getOperand(1));
     break;
   case RISCV::C_J:
-    // c.j $imm -> jal X0, $imm.
-    Res.setOpcode(RISCV::JAL);
-    Res.addOperand(MCOperand::createReg(RISCV::X0));
-    Res.addOperand(Inst.getOperand(0));
+    // c.j $imm -> (c)jal X0, $imm.
+    if (IsCapMode) {
+      Res.setOpcode(RISCV::CJAL);
+      Res.addOperand(MCOperand::createReg(RISCV::C0));
+      Res.addOperand(Inst.getOperand(0));
+    } else {
+      Res.setOpcode(RISCV::JAL);
+      Res.addOperand(MCOperand::createReg(RISCV::X0));
+      Res.addOperand(Inst.getOperand(0));
+    }
```

Comment:
```suggestion
    // c.j $imm -> (c)jal X0, $imm.
    Res.setOpcode(IsCapMode ? RISCV::CJAL : RISCV::JAL);
    Res.addOperand(
      MCOperand::createReg(IsCapMode ? RISCV::C0 : RISCV::X0));
    Res.addOperand(Inst.getOperand(0));
```
Maybe like this to make it a bit shorter?

Reply:
It's shorter but I find it harder to read and easier to mess up like that given the condition is duplicated, at least for me

Reply:
Though I guess I'm inconsistent as I did that for the expansion of (C)CALL